deploy/azuredeploy.bicep

```diff
@@ -197,201 +175,277 @@ resource blobService 'Microsoft.Storage/storageAccounts/blobServices@2021-04-01'
   properties: {
-    value: listkeys(iotIngestAuthorizationRule.id, iotIngestAuthorizationRule.apiVersion).primaryConnectionString
+    value: listkeys(en_basename_ingest_FunctionSender.id, '2021-01-01-preview').primaryConnectionString
   }
-  dependsOn: [
-    infraKeyVault
-  ]
 }
 
-resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
+resource la_basename 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
   name: 'la-${basename}'
-  location: resourceGroup().location
+  location: location
   properties: {
     sku: {
       name: 'PerGB2018'
     }
   }
 }
 
-resource appInsights 'Microsoft.Insights/components@2020-02-02' = {
+resource ai_basename 'Microsoft.Insights/components@2020-02-02' = {
   name: 'ai-${basename}'
-  location: resourceGroup().location
+  location: location
   kind: 'web'
   properties: {
     Application_Type: 'web'
-    WorkspaceResourceId: logAnalyticsWorkspace.id
+    WorkspaceResourceId: la_basename.id
   }
 }
 
-resource hostingPlan 'Microsoft.Web/serverfarms@2020-10-01' = {
+resource app_plan_basename 'Microsoft.Web/serverfarms@2020-10-01' = {
   name: 'app-plan-${basename}'
-  location: resourceGroup().location
+  location: location
   sku: {
     name: 'Y1'
     tier: 'Dynamic'
+    size: 'Y1'
+    family: 'Y'
+    capacity: 0
+  }
+  kind: 'functionapp'
+  properties: {
+    perSiteScaling: false
+    maximumElasticWorkerCount: 1
+    isSpot: false
+    reserved: false
+    isXenon: false
+    hyperV: false
+    targetWorkerCount: 0
+    targetWorkerSizeId: 0
   }
 }
 
-resource identityFn 'Microsoft.Web/sites@2020-06-01' = {
+resource identity_basename 'Microsoft.Web/sites@2020-06-01' = {
   name: 'identity-${basename}'
-  location: resourceGroup().location
+  location: location
   kind: 'functionapp'
   identity: {
     type: 'SystemAssigned'
   }
   properties: {
+    enabled: true
     httpsOnly: true
-    serverFarmId: hostingPlan.id
+    serverFarmId: app_plan_basename.id
+    reserved: false
+    scmSiteAlsoStopped: false
     clientAffinityEnabled: true
-    siteConfig: {
-      appSettings: union(commonAppSettings, [
-        {
-          name: 'WEBSITE_CONTENTSHARE'
-          value: 'identity-${basename}-${take(uniqueString('identity-', basename), 4)}'
-        }
-        {
-          name: 'GOOGLE_OAUTH_CLIENT_ID'
-          value: google_client_id
-        }
-        {
-          name: 'GOOGLE_OAUTH_CLIENT_SECRET'
-          value: google_client_secret
-        }
-        {
-          name: 'USERS_KEY_VAULT_URI'
-          value: 'https://${usersKvName}${environment().suffixes.keyvaultDns}'
-        }
-      ])
-    }
+    clientCertEnabled: false
+    hostNamesDisabled: false
+    containerSize: 1536
+    dailyMemoryTimeQuota: 0
+  }
+}
+
+resource identity_basename_appsettings 'Microsoft.Web/sites/config@2015-08-01' = {
+  parent: identity_basename
+  location: location
+  name: 'appsettings'
+  properties: {
+    FUNCTIONS_EXTENSION_VERSION: '~3'
```

Comment:
Suggestion: Might want to look at switching to v4 for .NET 6 support.

Reply:
I think this is a good idea, however it will require some changes outside the scope of this PR. I will submit a followup PR to support .NET 6.